### PR TITLE
Postmaster (Server + Client)

### PR DESCRIPTION
**Pull request by jarulraj**
A postmaster based on `asyncio` module.

- server (uses command_handler to process a request)
- command_handler
- client (uses interpreter)
- interpreter
- test_server
- test_client

### CODE REVIEW

File: src/server/server.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+    # These counters are used for realtime server monitoring
+    __connections__ = 0
+    __errors__ = 0
+
+    def connection_made(self, transport):
+        self.transport = transport
+
+        # Set timeout for sockets
+        if not set_socket_io_timeouts(self.transport, 60, 0):
```
Comment (saiprashanth173) [inline src/server/server.py]: Shouldn't the timeouts be configurable? 

Maybe, we can use them as attributes in EvaServer? 
eg. EvaServer(timeout_secs=...). By doing this we can read the config only when EvaServer in instantiated.

File: test/server/test_client.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        def timeout_server():
+            # need a more robust mechanism for when to cancel the future
+            time.sleep(2)
+            self.stop_clients_future.cancel()
+
+        thread = threading.Thread(target=timeout_server)
+        thread.daemon = True
+        thread.start()
+
+        summary = start_clients(client_count=client_count,
```
Comment (saiprashanth173) [inline test/server/test_client.py]: Should we have some assert statements here?

File: test/server/test_interpreter.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+from src.server.interpreter import EvaCommandInterpreter
+
+
+class InterpreterTests(unittest.TestCase):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def test_run_event_loop(self):
+
```
Comment (saiprashanth173) [inline test/server/test_interpreter.py]: I assume these tests are to be completed when we finish the interpreter implementation.

File: src/utils/logging_manager.py [hunk 4]
```diff
@@ -60,6 +61,23 @@ def log(self, string, level: LoggingLevel = LoggingLevel.DEBUG):
         elif level == LoggingLevel.CRITICAL:
             self._LOG.critical(string)
 
+    def setEffectiveLevel(self, level: LoggingLevel):
+
+        # Note: pytest logging level cannot be higher than WARNING
+        # https://github.com/segevfiner/pytest/blob/master/
+        # _pytest/logging.py#L246
+
+        if level == LoggingLevel.DEBUG:
```
Comment (saiprashanth173) [inline src/utils/logging_manager.py]: Why do we need these if else conditions? 

Can we use : self._LOG.setLevel(level.value) instead?

File: test/server/test_server.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+                         stop_server_future=self.stop_server_future)
+
+    def test_server_protocol(self):
+
+        eva_server = EvaServer()
+        eva_server.transport = mock.Mock()
+        eva_server.transport.close = MagicMock(return_value="closed")
+        eva_server.transport.abort = MagicMock(return_value="aborted")
+
+        # connection made
```
Comment (saiprashanth173) [inline test/server/test_server.py]: I feel this test should be split into multiple tests.

File: src/server/server.py (hunk 1 above)
Comment (jarulraj) [reply src/server/server.py]: Yes, added it to the eva.yml file.

File: test/server/test_client.py (hunk 2 above)
Comment (jarulraj) [reply test/server/test_client.py]: Done.

File: test/server/test_interpreter.py (hunk 3 above)
Comment (jarulraj) [reply test/server/test_interpreter.py]: Yes, we need to expand it.

File: src/utils/logging_manager.py (hunk 4 above)
Comment (jarulraj) [reply src/utils/logging_manager.py]: Good idea, done.

File: test/server/test_server.py (hunk 5 above)
Comment (jarulraj) [reply test/server/test_server.py]: Done.

**Pull request merged**